### PR TITLE
mimalloc: add version 2.0.3

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **mimalloc/2.0.3**

This version includes several performance improvements which is the most reason to use mimalloc.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mimalloc/all/conanfile.py [hunk 1]
```diff
@@ -142,6 +143,7 @@ def package(self):
             cmake.install()
 
         tools.rmdir(os.path.join(self.package_folder, "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```
Comment (prince-chrismc) [inline recipes/mimalloc/all/conanfile.py:146]: Looks like cmake names are not set correctly https://github.com/microsoft/mimalloc/blob/0560fc27c08d28d523b7f741a42deb26cd01c0c6/CMakeLists.txt#L332

Comment (toge) [reply recipes/mimalloc/all/conanfile.py:146]: I'm sorry. I don't know what to do.
If you don't mind, could you tell me what to fix?

Comment (prince-chrismc) [reply recipes/mimalloc/all/conanfile.py:146]: Whenever I see changes about the CMake config files, my reaction it to look at the changes in the upstream `CMakeLists.txt` ... they fixed up the install dirs but also installed the targets 

You need to configure the names in `package_info` to match

```py
        self.cpp_info.names["cmake_find_package"] = "mimalloc" if self.options.shared else "mimalloc-static"
        self.cpp_info.names["cmake_find_package_multi"] = "mimalloc" if self.options.shared else "mimalloc-static"
```

Comment (toge) [reply recipes/mimalloc/all/conanfile.py:146]: Thank you very much!
Now I understand your point well.

I will remember your point when I update some of the recipes.
My understanding of conan is improved one more.

Comment (toge) [reply recipes/mimalloc/all/conanfile.py:146]: Fixed

**Review by prince-chrismc — APPROVED** [review]

**Review by ericriff — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**